load-generator/Dockerfile

```diff
@@ -10,7 +10,7 @@ COPY ./build/ ./build/
 # default 1.0
 ARG VER=1.0
 
-RUN gradle build
+# assume app is already built since it will not build in docker container
```

Comment:
Why?

Reply:
Yes, why?

Also, if not building the application in the Dockerfile then there's more extensive surgery to do.  This probably doesn't even need to be a multi-stage build if that's the case.

Reply:
[trace-java-client](https://github.com/aws-observability/aws-otel-test-framework/tree/terraform/trace-java-client) is referenced as local project by two gradle projects: load-generator and [jaeger-zipkin-sample-app](jaeger-zipkin-sample-app). 
`Load-generator` build by `Gradle` works by referencing to `trace-java-client` local project. But to build `load-generator` inside `dockerfile` requires commenting out [line 57](https://github.com/aws-observability/aws-otel-test-framework/blob/06496c5e679991d839b628afd737dcf28ce35214/load-generator/build.gradle#L57) and uncommenting [line 54 to 56](https://github.com/aws-observability/aws-otel-test-framework/blob/06496c5e679991d839b628afd737dcf28ce35214/load-generator/build.gradle#L54). - This is the current setup. Dockerfile and Gradle project are not able to build sameway b/c of difference in referencing a local project.

So, I noticed that `jaeger-zipkin-sample-app` resolves this thing by [simply not building project inside dockerfile]( https://github.com/aws-observability/aws-otel-test-framework/blob/06496c5e679991d839b628afd737dcf28ce35214/sample-apps/jaeger-zipkin-sample-app/Dockerfile#L12) and rather requires that the project is built before the dockerfile is built so that Docker can simply copy the executable over.

Now with this, the first stage of copying the source code is indeed meaningless as we are expecting the binary to be available in the local filesystem before docker build command. I avoided to have this change in this PR, but if it's approved then I can edit the dockerfile of both these projects in this PR.

Reply:
Have you taken a look into the Java sample apps? They use jib to build and push the container images.

https://github.com/aws-observability/aws-otel-java-instrumentation/blob/main/sample-apps/spark/build.gradle.kts#L27


If possible we should avoid assuming things and just build everything.

There is an example here of how to use job in a multi project environment
https://github.com/GoogleContainerTools/jib/blob/master/examples/multi-module/name-service/build.gradle#L19

Reply:
Thanks @rapphil for this suggestion. I was not aware of this tool, but now I see it quite a lot in test-framework. I have updated the code to incorporate this. Since the `dockerfile` is not used anymore so I have removed it.